.github/workflows/rust.yml

```diff
@@ -234,7 +234,7 @@ jobs:
           rust-version: stable
       - name: Run cargo doc
         run: |
-          export RUSTDOCFLAGS="-D warnings -A rustdoc::private-intra-doc-links"
+          export RUSTDOCFLAGS="-D warnings"
```

Comment:
this makes all doc warnings an error and does not allow links to private items

Reply:
`RUSTDOCFLAGS` in `ci/scripts/rust_docs.sh` should be updated as well
i am doing this in https://github.com/apache/datafusion/pull/11890